packages/grid/src/elements/splitter/PaneProvider.tsx

```diff
@@ -0,0 +1,260 @@
+/**
+ * Copyright Zendesk, Inc.
+ *
+ * Use of this source code is governed under the Apache License, Version 2.0
+ * found at http://www.apache.org/licenses/LICENSE-2.0.
+ */
+
+import React, { useState, useMemo, useCallback } from 'react';
+import PropTypes from 'prop-types';
+import { SplitterContext } from '../../utils/useSplitterContext';
+import { DIMENSIONS, UNITS, IPaneProvider } from '../../types';
+
+const getPixelsPerFr = (totalFrs: number, totalDimension: number) => {
+  return totalDimension / totalFrs;
+};
+
+const convertToPixels = (values: Record<string, number>, pixelsPerFr: number) => {
+  return Object.entries(values).reduce((prev, [key, value]) => {
+    prev[key] = value * pixelsPerFr;
+
+    return prev;
+  }, {} as Record<string, number>);
+};
+
+const PaneProviderComponent = ({
+  totalPanesWidth,
+  totalPanesHeight,
+  defaultRowValues,
+  defaultColumnValues,
+  rowValues,
+  columnValues,
+  onChange,
+  children
+}: IPaneProvider) => {
+  const isControlled =
+    rowValues !== undefined &&
+    rowValues !== null &&
+    columnValues !== undefined &&
+    columnValues !== null;
+  const [rowState, setRowState] = useState<Record<string, number>>(defaultRowValues || {});
+  const [columnState, setColumnState] = useState<Record<string, number>>(defaultColumnValues || {});
+  const rowsTrack = isControlled ? rowValues : rowState;
+  const columnsTrack = isControlled ? columnValues : columnState;
+
+  const setRowsTrack = useCallback(
+    (values: (state: Record<string, number>) => Record<string, number>) => {
+      if (onChange) {
+        return onChange(values(rowsTrack), columnsTrack);
+      }
+
+      return setRowState(values);
+    },
+    [onChange, setRowState, columnsTrack, rowsTrack]
+  );
+
+  const setColumnsTrack = useCallback(
+    (values: (state: Record<string, number>) => Record<string, number>) => {
+      if (onChange) {
+        return onChange(rowsTrack, values(columnsTrack));
+      }
+
+      return setColumnState(values);
+    },
+    [onChange, setColumnState, rowsTrack, columnsTrack]
+  );
+
+  const totalFractions = useMemo(
+    () => ({
+      rows: Object.values(rowsTrack).reduce((prev, value) => value + prev, 0),
+      columns: Object.values(columnsTrack).reduce((prev, value) => value + prev, 0)
+    }),
+    [rowsTrack, columnsTrack]
+  );
+
+  const pixelsPerFr = useMemo(
+    () => ({
+      rows: getPixelsPerFr(totalFractions.rows, totalPanesHeight),
+      columns: getPixelsPerFr(totalFractions.columns, totalPanesWidth)
+    }),
+    [totalFractions, totalPanesHeight, totalPanesWidth]
+  );
+
+  const layoutStateInPixels = useMemo(
+    () => ({
+      rows: convertToPixels(rowsTrack, pixelsPerFr.rows),
+      columns: convertToPixels(columnsTrack, pixelsPerFr.columns)
+    }),
+    [rowsTrack, columnsTrack, pixelsPerFr]
+  );
```

Comment:
Could these memos be combined?
```
const { totalFractions, pixelsPerFr, layoutStateInPixels } = useMemo(() => ({ ... }))
```

Reply:
After refactoring, I prefer the original method for its simplicity. Here is what a collapsed version looks like:

<img width="792" alt="Screen Shot 2022-04-28 at 4 49 51 PM" src="https://user-images.githubusercontent.com/1975639/165865348-16613355-616b-4905-966a-7938f89945b7.png">